src/main/java/com/atlassian/db/replica/internal/ReplicaConnectionProvider.java

```diff
@@ -80,7 +82,7 @@ public boolean getReadOnly() {
     public void setReadOnly(boolean readOnly) throws SQLException {
         isReadOnly = readOnly;
         if (readOnly) {
-            state.getReadConnection().setReadOnly(isReadOnly);
+            state.getReadConnection(QueryContext.builder(Reason.API_CALL,false)).setReadOnly(isReadOnly);
```

Comment:
Actually, if we add a "cause" to `API_CALL`, then we can reuse `API_CALL`

Reply:
Any idea why would we care about this level of details? Any example of a use case?

Reply:
We could see we use main due to too many `executeUpdate` calls or too many `executeBatch`. We could then focus on usages of that specific method in Jira. There are a lot of signatures of methods with those names, so this amount of detail could indeed be unnecessary. If we assume that these calls are unchangeable, then indeed the drilldown doesn't help.

Reply:
IMO we need to distinguish between RO_API calls and MAIN_API calls, for now, we can add more details later if needed (and I don't think it will be needed if we mix it with stack-trace mechanism).

Reply:
Agreed